scripts/trn_condensed.R

```diff
@@ -0,0 +1,129 @@
+  rowwise |>
+  mutate(pair_id = paste(sort(c(trn1, trn2)), collapse = "_")) |>
+  ungroup()
+
+# Group and merge publication and trn1inreg2/trn2inreg1 information
+trn_merged_pubs <- trn_trn_pair_id %>%
+  group_by(pair_id) %>%
+
+  # Copy the trn2 boolean information of the other trial with the same pair_id into the other trials trn1 boolean fields
+  mutate(
+    trn1_in_pub_si = if_else(trn1 == first(trn1), lead(trn2_in_pub_si), lag(trn2_in_pub_si)),
+    trn1_in_pub_abs = if_else(trn1 == first(trn1), lead(trn2_in_pub_abs), lag(trn2_in_pub_abs)),
+    trn1_in_pub_ft = if_else(trn1 == first(trn1), lead(trn2_in_pub_ft), lag(trn2_in_pub_ft)),
+
+    trn1inreg2 = case_when(
+      trn1 == first(trn1) & (is.na(trn1inreg2) & is.na(lead(trn2inreg1))) ~ NA,
+      trn1 == first(trn1) & (trn1inreg2 == lead(trn2inreg1)) ~ trn1inreg2,
+      trn1 == first(trn1) & (is.na(trn1inreg2) | is.na(lead(trn2inreg1))) ~ coalesce(trn1inreg2, lead(trn2inreg1), FALSE),
+
+      trn1 != first(trn1) & (is.na(trn1inreg2) & is.na(lag(trn2inreg1))) ~ NA,
+      trn1 != first(trn1) & (trn1inreg2 == lag(trn2inreg1)) ~ trn1inreg2,
+      trn1 != first(trn1) & (is.na(trn1inreg2) | is.na(lag(trn2inreg1))) ~ coalesce(trn1inreg2, lag(trn2inreg1), FALSE),
+
+      .default = NA
+    ),
+    trn2inreg1 = case_when(
+      trn1 == first(trn1) & (is.na(trn2inreg1) & is.na(lead(trn1inreg2))) ~ NA,
+      trn1 == first(trn1) & (trn2inreg1 == lead(trn1inreg2)) ~ trn2inreg1,
+      trn1 == first(trn1) & (is.na(trn2inreg1) | is.na(lead(trn1inreg2))) ~ coalesce(trn2inreg1, lead(trn1inreg2), FALSE),
+
+      trn1 != first(trn1) & (is.na(trn2inreg1) & is.na(lag(trn1inreg2))) ~ NA,
+      trn1 != first(trn1) & (trn2inreg1 == lag(trn1inreg2)) ~ trn2inreg1,
+      trn1 != first(trn1) & (is.na(trn2inreg1) | is.na(lag(trn1inreg2))) ~ coalesce(trn2inreg1, lag(trn1inreg2), FALSE),
+
+      .default = NA
+    )
+  ) %>%
+  ungroup()
+
+# Merge protocol ID and title matching information
+trn_protocols_merged <- trn_merged_pubs |>
+  group_by(pair_id) |>
+  mutate(
+    is_match_protocol_sponsor_protocol_id = ifelse(any(!is.na(is_match_protocol_sponsor_protocol_id)),
+                                                   TRUE, NA),
+    is_match_results_sponsor_protocol_id = ifelse(any(!is.na(is_match_results_sponsor_protocol_id)),
+                                                  TRUE, NA),
+    is_title_matched = ifelse(any(!is.na(is_title_matched)),
+                              TRUE, NA)) |>
+  ungroup()
+
+# Merge meta booleans
+trn_merged_meta <- trn_protocols_merged |>
+  group_by(pair_id) |>
+  mutate(
+    at_least_one_pub = ifelse(any(at_least_one_pub), TRUE, FALSE),
+    at_least_one_sponsor_match = ifelse(any(at_least_one_sponsor_match), TRUE, FALSE)) |>
+  ungroup()
+
+# Drop duplicated row. They should have the same information now, so just drop the first.
+trn_no_dupes <- trn_merged_meta |>
+  group_by(pair_id) |>
+  slice_head(n= 1) |>
+  ungroup()
+
```

Comment:
To finish the above alternative, merge the cleaned duplicates (ranks_merged) and the non-duplicates. The final steps (priorities and cleanup) can be ran on this data frame without any further changes.
```suggestion
trn_no_dupes_ranks <- trn_trn_pair_id |>
  filter(!pair_id %in% ranks_merged$pair_id) |> # remove old dupes first
  bind_rows(ranks_merged) |> # add deduplicated trns
  arrange(trn1, trn2)
```